### PR TITLE
pkcs1: implement support for PSS Params structure

### PR DESCRIPTION
**Pull request by lumag**
Add support for parsing and generatign RSA PSS params.

Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@linaro.org>

### CODE REVIEW

File: pkcs1/src/params.rs [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+const SHA_1_AI: AlgorithmIdentifier<'_> = AlgorithmIdentifier {
+    oid: OID_SHA_1,
+    parameters: None,
+};
+
+fn default_mgf1_sha1() -> AlgorithmIdentifier<'static> {
+    AlgorithmIdentifier {
+        oid: OID_MGF_1,
+        parameters: Some(
+            AnyRef::new(Tag::Sequence, &[0x06, 0x5, 0x2b, 0xe, 0x3, 0x2, 0x1a]).unwrap(),
```
Comment (tarcieri) [inline pkcs1/src/params.rs:46]:
```suggestion
            AnyRef::from(&OID_SHA_1)
```

Comment (lumag) [reply pkcs1/src/params.rs:46]: Not quite. This will end up with the Any containing just `OID` rather than `SEQUENCE` with the single `OID` inside

Comment (lumag) [reply pkcs1/src/params.rs]:
```
   0  48: SEQUENCE {
  2  13:   [0] {
  4  11:     SEQUENCE {
  6   9:       OBJECT IDENTIFIER sha-256 (2 16 840 1 101 3 4 2 1)
       :       }
       :     }
 17  26:   [1] {
 19  24:     SEQUENCE {
 21   9:       OBJECT IDENTIFIER pkcs1-MGF (1 2 840 113549 1 1 8)
 32  11:       SEQUENCE {
 34   9:         OBJECT IDENTIFIER sha-256 (2 16 840 1 101 3 4 2 1)
       :         }
       :       }
       :     }
 45   3:   [2] {
 47   1:     INTEGER 32
       :     }
       :   }
```

Comment (tarcieri) [reply pkcs1/src/params.rs]: Aah. Annoying

**Comment by lumag** [comment]
@tarcieri simplified the implementation to use `derive[Sequence]`. However I still could not simplify the `AnyRef` part.

File: pkcs1/src/params.rs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+///     saltLength         [2] INTEGER            DEFAULT 20,
+///     trailerField       [3] TrailerField       DEFAULT trailerFieldBC
+/// }
+/// HashAlgorithm ::= AlgorithmIdentifier
+/// MaskGenAlgorithm ::= AlgorithmIdentifier
+/// ```
+///
+/// [RFC 8017 Appendix 2.3]: https://datatracker.ietf.org/doc/html/rfc8017#appendix-A.2.3
+#[derive(Clone, Debug, Eq, PartialEq, Sequence)]
+pub struct RsaPSSParameters<'a> {
```
Comment (tarcieri) [inline pkcs1/src/params.rs:71]: Per [RFC 430 naming conventions](https://rust-lang.github.io/api-guidelines/naming.html), this should be:

```suggestion
pub struct RsaPssParameters<'a> {
```

**Review by tarcieri — CHANGES_REQUESTED** [review]
One nit otherwise looks good

**Review by tarcieri — APPROVED** [review]
I'll go ahead and fix up the name myself

**Pull request merged**